### PR TITLE
[MDEPLOY-314] Include artifactId in DeployMojo#processProject messages

### PR DESCRIPTION
**Pull request by apeteri**
The PR addresses [MDEPLOY-314](https://issues.apache.org/jira/browse/MDEPLOY-314) by including the artifact ID of the currently processed project in `MojoExecutionException` and warning messages.

---

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEPLOY) filed for the change (usually before you start working on it). Trivial changes like typos do not require a JIRA issue. Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEPLOY-XXX] - Fixes bug in ApproximateQuantiles`, where you replace `MDEPLOY-XXX` with the appropriate JIRA issue. Best practice is to use the JIRA issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by cstamas — APPROVED** [review]
Thanks!

**Comment by cstamas** [comment]
merged w/ tweaks as b9c1c8bc486c7defb92171e1ea9169075b76695c

File: src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java [hunk 1]
```diff
@@ -299,18 +300,19 @@ private void processProject(final MavenProject project, DeployRequest request) t
...
-                    throw new MojoExecutionException("The packaging plugin for this project did not assign "
-                            + "a main file to the project but it has attachments. Change packaging to 'pom'.");
+                    throw new MojoExecutionException("The packaging plugin for project " + project.getArtifactId()
+                            + " did not assign a main file to the project but it has attachments. Change packaging"
+                            + " to 'pom'.");
                 }
             } else {
-                throw new MojoExecutionException(
-                        "The packaging for this project did not assign a file to the build artifact");
+                throw new MojoExecutionException("The packaging plugin for project " + project.getArtifactId()
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java:314]: "packaging plugin"?
there is no packaging plugin, just packaging

File: src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java [hunk 2]
```diff
@@ -299,18 +300,19 @@ private void processProject(final MavenProject project, DeployRequest request) t
             } else if (!project.getAttachedArtifacts().isEmpty()) {
                 if (allowIncompleteProjects) {
                     getLog().warn("");
-                    getLog().warn("The packaging plugin for this project did not assign");
+                    getLog().warn("The packaging plugin for project " + project.getArtifactId() + " did not assign");
```
Comment (hboutemy) [inline src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java:303]: same "packaging plugin" issue

**Comment by jira-importer** [comment]
Resolve #407

**Pull request closed without merging**